### PR TITLE
Add --glob and --regex

### PR DESCRIPTION
**Pull request by iology**
WIP:

- ~~[ ] update the "globset" crate in the upstream~~?
- [ ] documentation
- [ ] the behavior of "/" and "\\" on Windows file paths (BurntSushi/ripgrep#526)
- [x] test with `--full-path`
- [ ] more tests
- [ ] todos and fixme in the source code

ping #85

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -123,9 +125,19 @@ fn main() {
...
 
-    match RegexBuilder::new(pattern)
-              .case_insensitive(!config.case_sensitive)
-              .build() {
+    let mut builder = if !matches.is_present("use-glob") {
+        RegexBuilder::new(pattern)
+    } else {
+        match GlobBuilder::new(pattern).literal_separator(true).build() {
+            Ok(glob) => {
+                // TODO: provide an option to remove (?-u) in the upstream?
```
Comment (sharkdp) [inline src/main.rs]: is this still relevant?

File: src/main.rs [hunk 2]
```diff
@@ -123,9 +125,19 @@ fn main() {
...
+                eprintln!("GLOB:{} | REGEX: {}", glob.glob(), glob.regex());
```
Comment (sharkdp) [inline src/main.rs]: Is this a debug print?

Comment (iology) [reply src/main.rs]: Yes, just to get you familiar with how --glob works.

File: src/main.rs (hunk 1 above)
Comment (iology) [reply src/main.rs]: If it is unnecessary to match arbitrary bytes in the path, then no need to remove (?-u).

**Comment by iology** [comment]
Added tests for --full-path and rebased.

**Comment by iology** [comment]
Another problem is how to apply `(?s)` (allow `.` to match newlines) to glob patterns.

**Comment by sharkdp** [comment]
> Another problem is how to apply (?s) (allow . to match newlines, "dot-all") to glob patterns.

I think we should do the same as for #111.

**Comment by iology** [comment]
Yes, and no option to disable "dot-all" for glob patterns? I'm ok with this.

**Comment by sharkdp** [comment]
> Yes, and no option to disable "dot-all" for glob patterns? I'm ok with this.

Me too.

**Comment by iology** [comment]
I've added some todos in the top comment.

**Comment by iology** [comment]
Closing. I plan to do this in my private fork. Thanks for making `fd`. It really helps to get me familiar with Rust.

**Pull request closed without merging**